ecl/eclccserver/eclccserver.cpp

```diff
@@ -602,18 +612,29 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
                 removeGeneratedFiles(wuid);
 #endif
                 queryDllServer().registerDll(realdllname.str(), "Workunit DLL", dllurl.str());
-                workunit->commit();
-                return true;
             }
             else
             {
 #ifndef _CONTAINERIZED
                 Owned<IWUQuery> query = workunit->updateQuery();
                 associateLocalFile(query, FileTypeLog, logfile, "Compiler log", 0);
-                workunit->commit();
 #endif
-                return false;
             }
+
+            if (compileCppSeparately)
+            {
+                //Files need to be added after the workunit has been cloned - otherwise they are overwritten
+                Owned<IWUQuery> query = workunit->updateQuery();
+                VStringBuffer ccfileName("%s.cc", wuid);
+                VStringBuffer cclogfileName("%s.cc.log", wuid);
+                if (checkFileExists(cclogfileName))
+                    associateLocalFile(query, FileTypeLog, cclogfileName, "CPP log", 0);
+                if (saveTemps && checkFileExists(ccfileName))
+                    associateLocalFile(query, FileTypeLog, ccfileName, "compile actions log", 0);
```

Comment:
If saveTemps not set should delete it?

Reply:
if save temps not set it has already been deleted.

Reply:
but if the eclcc failed it still may not exists - hence the check